src/App.tsx

```diff
@@ -1,223 +1,230 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import {
+  getTodos,
+  addTodo,
+  patchTodo,
+  removeTodo,
+} from './api/todos';
+
 import { AuthContext } from './components/Auth/AuthContext';
 
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
+
+import { FilterBy } from './types/FilterBy';
+import { Todo, TodoToPost } from './types/Todo';
+import { User } from './types/User';
+
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const user = useContext(AuthContext);
+  const user: User | null = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [isSuccessful, setIsSuccessful] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.All);
+
+  const [isAdding, setIsAdding] = useState(false);
+
+  const completedTodos = useMemo(() => (
+    todos.filter(({ completed }) => completed)
+  ), [todos]);
+
+  const getTodosFromApi = async () => {
+    try {
+      if (user) {
+        const todosFromApi = await getTodos(user.id);
+
+        setTodos(todosFromApi);
+        setVisibleTodos(todosFromApi);
+      }
+    } catch (error) {
```

Comment:
```suggestion
    } catch {
```
Unused var. Fix everywhere